src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+import numpy as np
+import os
+import h5py
+import math
+import time
+import json
+import argparse
+import logging
+import multiprocessing
+import functools
+
+
+# Default parameters
+minvalue = 0.0
+maxlevel = 65535  # 16bit
+
+valid_nr_methods = [0, 1, 2, 3]
+
+
+def mode(data: np.ndarray):
+    """
+    Compute the mode of a numpy array.
+
+    Parameters
+    ----------
+    data: np.ndarray
+        Numpy array containing data of which we will calculate the mode.
+    Returns
+    -------
+    list
+        All values which appear the most number of times
+    int
+        The number of times the mode values appear
+    """
+    counts = {}
+
+    for x in data.flatten():
+        counts[x] = counts.get(x, 0) + 1
+
+    maxcount = max(counts.values())
+    modelist = []
+
+    for x in counts:
+        if counts[x] == maxcount:
+            modelist.append(x)
+
+    return modelist, maxcount
+
+
+def noise_reduce(data: np.array,
+                 xtable: dict,
+                 ind: int,
+                 xindex: np.array,
+                 noise_reduction: int):
+    """
+    Noise reduce a numpy array by one of several methods, specified by
+    the noise_reduction argument. The current acceptable values are:
+    - 0: No noise reduction
+    - 1: Reduce the data value by 2 * sigma
+    - 2: Reduce the data value by sigma and also normalize it
+    - 3: Normalize the data value
+
+    Parameters
+    ----------
+    data: np.ndarray
+        The data that will be noise reduced.
+    xtable: dict
+        A dict containing aggregate information about the input video
+        to be used in the dewarping process.
+    ind: int
+        The column currently being noise reduced.
+    xindex: np.array
+        Array of indices of the input image to be used for the output image.
+    noise_reduction: int
+        The noise reduction method to use.
+    Returns
+    -------
+    np.ndarray
+        The data, of the sme shape as the input data, having had the
+        noise reduction process applied to it
+    """
+    if 0 == noise_reduction:
+        pass
+
+    elif 1 == noise_reduction:
+        nf = (
+            2 * xtable['stdv_modevalue']
+            * (xindex[ind] - xindex[ind - 1] - 1)
+        )
+
+        data = data - nf
+
+    elif 2 == noise_reduction:
+        nf = 1 * xtable['stdv_modevalue'] * (xindex[ind] - xindex[ind - 1] - 1)
+        data = (data / (xindex[ind] - xindex[ind - 1])) - nf
+
+    elif 3 == noise_reduction:
+        data = data / (xindex[ind] - xindex[ind - 1])
+
```

Comment:
`else`? if it isn't [0, 1, 2, 3] then it returns the input data. The docstring should at least say that.